projectHashProposal.md

```diff
@@ -0,0 +1,18 @@
+# LSIF: ProjectWithHash
+
+Formulating and processing LSIF for large repositories can be very time consuming. We've discovered that many commits touch only ~20% of projects in a given repository, so there can be lots of unnecessary processing done on unchanged projects that are then ignored at ingestion time. To decrease processing times, I propose adding a new "projectHash" property to the existing [project vertex](https://github.com/Microsoft/language-server-protocol/blob/master/indexFormat/specification.md#the-project-vertex). The new definition for the `project` vertex would be:
+
+```ts
+export interface ProjectWithHash extends Project {
```

Comment:
Why not just make it part of the Project interface since it is optional already.

Reply:
Sounds good! I added the optional projectHash to the existing LSIF spec, but kept this doc for now to facilitate discussion - I don't know if we want this much detail to enter the LSIF spec for this single property.